________________________________________________________________________________________________________________________


* Yuvraj Sutar 
* yuvrajsutar009@gmail.com
_ _________________________________________________________________________________________________________________________
_  _________________________________________________________________________________________________________________________


In [1]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
import matplotlib.colors as colors
import seaborn as sns
import itertools
from scipy.stats import norm
from sklearn.datasets import load_iris
import scipy.stats
from sklearn.naive_bayes import BernoulliNB
from sklearn.naive_bayes import MultinomialNB
from sklearn.naive_bayes import GaussianNB



%matplotlib inline
sns.set()

Load the data set


In [2]:
iris = pd.read_csv('C:\\Users\\Yuvraj\\Desktop\\Github\\Data\\iris.csv')
iris.head()

sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa

In [3]:
X = iris.iloc[:,0:4]
y= iris.iloc[:,4]

In [6]:
X

sepal_length  sepal_width  petal_length  petal_width
0             5.1          3.5           1.4          0.2
1             4.9          3.0           1.4          0.2
2             4.7          3.2           1.3          0.2
3             4.6          3.1           1.5          0.2
4             5.0          3.6           1.4          0.2
..            ...          ...           ...          ...
145           6.7          3.0           5.2          2.3
146           6.3          2.5           5.0          1.9
147           6.5          3.0           5.2          2.0
148           6.2          3.4           5.4          2.3
149           5.9          3.0           5.1          1.8

[150 rows x 4 columns]

In [7]:
y

0         setosa
1         setosa
2         setosa
3         setosa
4         setosa
         ...    
145    virginica
146    virginica
147    virginica
148    virginica
149    virginica
Name: species, Length: 150, dtype: object

Split dataset into training set and test set

In [8]:

from sklearn.model_selection import train_test_split 
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3,random_state=10000) 
X_train

sepal_length  sepal_width  petal_length  petal_width
20            5.4          3.4           1.7          0.2
123           6.3          2.7           4.9          1.8
137           6.4          3.1           5.5          1.8
17            5.1          3.5           1.4          0.3
109           7.2          3.6           6.1          2.5
..            ...          ...           ...          ...
2             4.7          3.2           1.3          0.2
47            4.6          3.2           1.4          0.2
40            5.0          3.5           1.3          0.3
53            5.5          2.3           4.0          1.3
140           6.7          3.1           5.6          2.4

[105 rows x 4 columns]

In [9]:
y_train

20         setosa
123     virginica
137     virginica
17         setosa
109     virginica
          ...    
2          setosa
47         setosa
40         setosa
53     versicolor
140     virginica
Name: species, Length: 105, dtype: object

## BernoulliNB Naive bayes

In [10]:
from sklearn import svm
from sklearn.naive_bayes import BernoulliNB

model1 = BernoulliNB()
#Train the model using the training sets
model1.fit(X_train, y_train)



BernoulliNB()

Predict the response for test dataset

In [11]:

y_pred = model1.predict(X_test)
y_pred

array(['versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor'], dtype='<U10')

Import scikit-learn metrics module for accuracy calculation

In [12]:

from sklearn import metrics

from sklearn.metrics import confusion_matrix, classification_report,accuracy_score

print("Accuracy BernoulliNB :",metrics.accuracy_score(y_test, y_pred))


Accuracy BernoulliNB : 0.2


### GaussianNB

In [13]:
from sklearn import svm
from sklearn.naive_bayes import GaussianNB

model2 = GaussianNB()
#Train the model using the training sets
model2.fit(X_train, y_train)




GaussianNB()

Predict the response for test dataset

In [14]:
y_pred = model2.predict(X_test)
y_pred

array(['virginica', 'virginica', 'setosa', 'virginica', 'versicolor',
       'virginica', 'versicolor', 'versicolor', 'virginica', 'virginica',
       'versicolor', 'versicolor', 'virginica', 'setosa', 'setosa',
       'virginica', 'virginica', 'versicolor', 'virginica', 'setosa',
       'virginica', 'virginica', 'setosa', 'virginica', 'setosa',
       'setosa', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'virginica', 'versicolor', 'virginica', 'virginica', 'virginica',
       'setosa', 'virginica', 'virginica', 'virginica', 'virginica',
       'setosa', 'virginica', 'virginica', 'versicolor', 'setosa'],
      dtype='<U10')

Import scikit-learn metrics module for accuracy calculation

In [77]:

from sklearn import metrics

from sklearn.metrics import confusion_matrix, classification_report,accuracy_score

print("Accuracy GaussianNB :",metrics.accuracy_score(y_test, y_pred))

Accuracy GaussianNB : 0.9555555555555556


In [60]:
con=confusion_matrix(y_test, y_pred)
con

array([[11,  0,  0],
       [ 0,  9,  0],
       [ 0, 25,  0]], dtype=int64)

## MultinomialNB Naive bayes

In [78]:
from sklearn import svm
from sklearn.naive_bayes import MultinomialNB

model3 = MultinomialNB()
#Train the model using the training sets
model3.fit(X_train, y_train)




MultinomialNB()

Predict the response for test dataset

In [79]:
y_pred = model3.predict(X_test)
y_pred

array(['versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor', 'setosa',
       'setosa', 'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'setosa', 'versicolor', 'versicolor', 'setosa', 'versicolor',
       'setosa', 'setosa', 'versicolor', 'setosa', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'setosa'], dtype='<U10')

Predict the response for test dataset

In [80]:

y_pred = model_sk.predict(X_test)
y_pred

array(['versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor', 'setosa',
       'setosa', 'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'setosa', 'versicolor', 'versicolor', 'setosa', 'versicolor',
       'setosa', 'setosa', 'versicolor', 'setosa', 'versicolor',
       'versicolor', 'versicolor', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'versicolor', 'setosa', 'versicolor', 'versicolor',
       'versicolor', 'setosa'], dtype='<U10')

Import scikit-learn metrics module for accuracy calculation

In [81]:
from sklearn import metrics

from sklearn.metrics import confusion_matrix, classification_report,accuracy_score

print("Accuracy MultinomialNB :",metrics.accuracy_score(y_test, y_pred))

Accuracy MultinomialNB : 0.4444444444444444


### Accuracy BernoulliNB : 0.2
### Accuracy GaussianNB : 0.9555555555555556
### Accuracy MultinomialNB : 0.4444444444444444

* Yuvraj Sutar 
* yuvrajsutar009@gmail.com